packages/client-utils/src/smart-wallet-with-sequence.ts

```diff
@@ -0,0 +1,182 @@
+/**
+ * @file Smart wallet wrapper with queue-based transaction sequence management
+ */
+
+import type { BridgeAction } from '@agoric/smart-wallet/src/smartWallet.js';
+import type { SignerData } from '@cosmjs/stargate';
+import { Fail } from '@endo/errors';
+import type {
+  SigningSmartWalletKit,
+  SmartWalletKit,
+} from './signing-smart-wallet-kit.js';
+import type { TxSequencer } from './sequence-manager.js';
+
+type QueuedThunk<T> = {
+  label: string;
+  resolve: (value: T) => void;
+  reject: (reason: unknown) => void;
+  thunk: () => Promise<T>;
+};
+
+const isSequenceNumberMismatch = (error: Error | unknown) => {
+  // @ts-expect-error reading "message" is fine here
+  const message = error?.message || `${error}`;
+  // *sigh*
+  return message.includes('account sequence mismatch');
+};
```

Comment:
I think we can do better, since ["account sequence mismatch" errors](https://github.com/agoric-labs/cosmos-sdk/blob/6ea4a76d5731effeb2594ac98dbcd8ec0a53556e/x/auth/ante/sigverify.go#L292) are associated with `sdkerrors.ErrWrongSequence` ([codespace "sdk" with number 32](https://github.com/agoric-labs/cosmos-sdk/blob/6ea4a76d5731effeb2594ac98dbcd8ec0a53556e/types/errors/errors.go#L106-L108), [propagated by cosmjs](https://github.com/cosmos/cosmjs/blob/b00ff7f9cde2d32f1fbbb91c160130c9f9c81a76/packages/stargate/src/stargateclient.ts#L490) and visible in our logs).
```suggestion
const isSequenceNumberMismatch = (error: BroadcastTxError | Error | unknown) =>
  // https://cosmos.github.io/cosmjs/latest/stargate/classes/BroadcastTxError.html
  error instanceof BroadcastTxError &&
  // https://pkg.go.dev/github.com/cosmos/cosmos-sdk/types/errors#pkg-variables:~:text=ErrWrongSequence
  error.codespace === 'sdk' &&
  error.code === 32;
```